packages/opentelemetry-tracer/package.json

```diff
@@ -52,6 +67,8 @@
     "@opentelemetry/core": "^0.0.1",
     "@opentelemetry/context-async-hooks": "^0.0.1",
     "@opentelemetry/scope-base": "^0.0.1",
-    "@opentelemetry/types": "^0.0.1"
+    "@opentelemetry/types": "^0.0.1",
+    "require-in-the-middle": "^4.0.0",
```

Comment:
Could you talk a bit more about why these are required? Given that this is a more generic package for the tracer, I'm nervous about adding Node-specific dependencies - or just dependencies in general that will add weight to the JS bundle when used in the browser.

Reply:
This is required to add hooks (pre and post middleware hooks to plugin methods), which allows us to modify modules on-the-fly as they are being required. I was thinking to have separate flavor of tracer (may be `web-tracer`) for browser. IIUC only `types` and `core` packages are common for both node and browser. WDYT?

Reply:
OK, would it make sense to call the package `node-tracer` then to make the distinction clear?

I was hoping that `basic-tracer` could at least be shared between the two as well.

Reply:
I believe the plugin API is only needed by the `automatic-tracer`, so we could maybe move it to there to be compatible with the browser ?

Reply:
IMO `Plugin` interface should stay in types and it makes sense to keep `PluginLoader` in node-tracer package.